src/reader.lisp

```diff
@@ -26,8 +44,10 @@ Used to forbid reading while inside quasiquoted forms.")
 
   (parser:with-reader-context stream
     (let* ((file
-             (coalton-impl/source:make-source-file *compile-file-truename*
-                                                   :offset (file-position stream)))
+             (coalton-impl/source:make-source-file (source-file)
```

Comment:
The source file backing the buffer may or may not be saved: the error annotations need to be based solely on the string passed to swank.